File: scripts/generate-padded-icons.mjs
#!/usr/bin/env node
/**
 * Build Safari/PWA icons with safe padding so rounded tiles don't clip the glow.
 * Run: node scripts/generate-padded-icons.mjs
 */
import path from "node:path";
import { fileURLToPath } from "node:url";
import sharp from "sharp";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const publicDir = path.resolve(__dirname, "../artifacts/shepherds-path/public");
const src = path.join(publicDir, "talk-it-through-icon.png");

/** Brand purple — matches theme-color / icon interior */
const BG = { r: 45, g: 20, b: 74, alpha: 1 };

/** Logo occupies ~72% of canvas (14% inset per side) */
const INSET = 0.14;

const OUTPUTS = [
  { size: 32, file: "favicon-32.png" },
  { size: 180, file: "apple-touch-icon.png" },
  { size: 192, file: "app-icon-192.png" },
  { size: 512, file: "app-icon-512.png" },
  { size: 1024, file: "sp-icon.png" },
];

async function renderIcon(size, outPath) {
  const inner = Math.max(8, Math.round(size * (1 - INSET * 2)));
  const logo = await sharp(src)
    .resize(inner, inner, { fit: "contain", background: BG })
    .png()
    .toBuffer();

  await sharp({
    create: {
      width: size,
      height: size,
      channels: 4,
      background: BG,
    },
  })
    .composite([{ input: logo, gravity: "center" }])
    .png()
    .toFile(outPath);
}

async function main() {
  for (const { size, file } of OUTPUTS) {
    const out = path.join(publicDir, file);
    await renderIcon(size, out);
    console.log(`Wrote ${file} (${size}×${size})`);
  }
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: scripts/generate-padded-icons.py
#!/usr/bin/env python3
"""Build Safari/PWA icons with safe padding so tiles don't clip the glow. Requires Pillow."""
from __future__ import annotations

import os
import shutil
from pathlib import Path

from PIL import Image

ROOT = Path(__file__).resolve().parents[1]
PUBLIC = ROOT / "artifacts" / "shepherds-path" / "public"
SRC = PUBLIC / "talk-it-through-icon.png"

# theme-color in index.html / manifest.json
BG = (59, 31, 122, 255)
INSET = 0.16

OUTPUTS = [
    (32, "favicon-32.png"),
    (180, "apple-touch-icon.png"),
    (192, "app-icon-192.png"),
    (512, "app-icon-512.png"),
    (1024, "sp-icon.png"),
]


def main() -> None:
    src = Image.open(SRC).convert("RGBA")
    for size, name in OUTPUTS:
        inner = max(8, int(size * (1 - INSET * 2)))
        logo = src.copy()
        logo.thumbnail((inner, inner), Image.Resampling.LANCZOS)
        canvas = Image.new("RGBA", (size, size), BG)
        x = (size - logo.width) // 2
        y = (size - logo.height) // 2
        canvas.paste(logo, (x, y), logo.split()[3])
        out = PUBLIC / name
        canvas.save(out, "PNG")
        print(f"Wrote {name} ({size}×{size})")

    shutil.copy(PUBLIC / "app-icon-512.png", PUBLIC / "app-icon.png")
    print("Synced app-icon.png")


if __name__ == "__main__":
    main()
